apps/web/src/components/PageEditor.tsx

```diff
@@ -48,6 +50,21 @@ export function PageEditor({ page, renderSubmit }: PageEditorProps) {
   const ceramic = useCeramic();
   const lit = useLit();
   const account = useAccount();
+  
+  const inputRef = useRef<HTMLInputElement>(null);
+  const router = useRouter();
+  const { query } = router;
+  const autofocusQuery = query.autofocus === "true";
```

Comment:
If it's possible it would be nice to avoid passing all these query parameters. I played around with it a bit on my local machine and it seems like the `AuthDialog` modal steals the focus. I found a fix for this that doesn't require query params, and looks like this:


1. wrap the ceramic session in a query
```tsx
  const ceramicSessionQuery = useQuery(
    [],
    async () => await ceramic.hasSession(),
    { cacheTime: 0 }
  );
```
2. Export loading states from `useCeramic` and `useLit`
```tsx
isLoading: isLitAuthenticatedQuery.isLoading,
...
isLoading: isComposeResourcesSignedQuery.isLoading,
```

3. Create computed loading prop
```tsx
  const isLoading =
    account.isConnecting ||
    ceramic.isLoading ||
    lit.isLoading ||
    ceramicSessionQuery.isLoading;
```

4. Only show modal if **not** loading
```tsx
      <AuthDialog open={!isLoading && !isAuthenticated} />
```

Reply:
And then just have this to focus the input field:
```tsx
  useEffect(() => {
      inputRef.current?.focus();
  }, []);
  ```

Reply:
This should eliminate the need for both storing state in the URL as well as passing the `autofocusProp` prop to the `PageEditor`

Reply:
This is better. I saw that the `AuthModal` did popup a few times, but I didn't know that it was due to that. Will change to your alternative.

Reply:
With the `cacheTime: 0` do we need to worry about rate limiting or anything like that?

Reply:
We already have:
```
  useEffect(() => {
    const run = async () =>
      setIsCeramicSessionValid(await ceramic.hasSession());
    run();
  }, [ceramic]);
```

But yeah we could use a query instead.

Don't think `cacheTime: 0` will cause any issues as it just disables caching and by default React Query tracks props passed accessed to not cause rerenders. @ericzakariasson can probably correct me 😅

Reply:
Yeah I replaced the `useEffect` @philipandersson  had with a `useQuery` to get the loading state for _free_. I set `cacheTime: 0` to replicate the initial behaviour, and I also think it's needed when you have an empty query key array!

Reply:
Let me know if you want me to push this code @Kenny-Gin1 !

Reply:
> Let me know if you want me to push this code @Kenny-Gin1 !

Yeah go ahead! I'll look at the code fixes - thanks for taking the time to find the real issue!

Reply:
I've pushed now!